test/unit/custom_index_test.lua

```diff
@@ -1,214 +1,240 @@
 local expirationd = require("expirationd")
 local t = require("luatest")
-local g = t.group("custom_index")
 
 local helpers = require("test.helper")
 
-g.before_all(function()
-    helpers.init_spaces(g)
+local g = t.group('custom_index', {
+    {index_type = 'TREE', engine = 'vinyl'},
+    {index_type = 'TREE', engine = 'memtx'},
+    {index_type = 'HASH', engine = 'memtx'},
+})
+
+g.before_each({index_type = 'TREE'}, function(cg)
+    g.space = helpers.create_space_with_tree_index(cg.params.engine)
+end)
+
+g.before_each({index_type = 'HASH'}, function(cg)
+    g.space = helpers.create_space_with_hash_index(cg.params.engine)
 end)
 
-g.after_each(function()
-    helpers.truncate_spaces(g)
+g.before_each({index_type = 'BITSET'}, function(cg)
+    g.space = helpers.create_space_with_bitset_index(cg.params.engine)
 end)
 
-function g.test_passing()
-    local task = expirationd.start("clean_all", g.tree.id, helpers.is_expired_true)
+g.after_each(function(g)
+    g.space:drop()
+end)
+
+function g.test_passing(cg)
+    t.skip_if(cg.params.index_type == 'BITSET', 'Unsupported index type')
+
+    local task = expirationd.start("clean_all", cg.space.id, helpers.is_expired_true)
     -- if we don't specify index, program should use primary index
-    t.assert_equals(task.index, box.space.tree.index[0])
+    t.assert_equals(task.index, cg.space.index[0])
     task:kill()
 
     -- index by name
-    task = expirationd.start("clean_all", g.tree.id, helpers.is_expired_true,
+    task = expirationd.start("clean_all", cg.space.id, helpers.is_expired_true,
             {index = "index_for_first_name"})
-    t.assert_equals(task.index, box.space.tree.index[1])
+    t.assert_equals(task.index, cg.space.index[1])
     task:kill()
 
     -- index by id
-    task = expirationd.start("clean_all", g.tree.id, helpers.is_expired_true,
+    task = expirationd.start("clean_all", cg.space.id, helpers.is_expired_true,
             {index = 1})
-    t.assert_equals(task.index, box.space.tree.index[1])
+    t.assert_equals(task.index, cg.space.index[1])
     task:kill()
+end
+
+function g.test_errors(cg)
+    t.skip_if(cg.params.index_type ~= 'TREE', 'Unsupported index type')
```

Comment:
Hm. Why can't you put this code into tree_errors? Anyway you run this test only for tree indexes.

Reply:
Agree.
```lua
--- a/test/unit/custom_index_test.lua
+++ b/test/unit/custom_index_test.lua
@@ -46,7 +46,7 @@ function g.test_passing(cg)
     task:kill()
 end
 
-function g.test_errors(cg)
+function g.test_tree_index_errors(cg)
     t.skip_if(cg.params.index_type ~= 'TREE', 'Unsupported index type')
 
     -- errors
```